src/app/icon_plugins.rs

```diff
@@ -0,0 +1,155 @@
+use std::collections::hash_map::HashMap;
+use crate::tree::TreeLineType;
+
+pub trait IconPlugin
+{
+	fn get_icon( 
```

Comment:
API to get Icon. To request an icon from a plugin, give it:

1) Entry type
2) File name: (so that stuff like LICENSE, Cargo.toml, some yarn etc files are distinguishable by full names)
3) Double extension (.tar.gz, etc)
4) Full path: Files can use libmagic, directories can list contents and determine what this is
5) extension